float32 obstacle_distance